cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -469,6 +532,31 @@ public RunId run(ProgramId programId, Map<String, String> overrides, boolean deb
     return runInternal(programId, userArgs, sysArgs, debug);
   }
 
+  /**
+   * Starts the latest version of a Program with the specified argument overrides.
+   *
+   * @param namespaceId the namespaceId of the program to run
+   * @param appName the application name of the program to run
+   * @param programType the program type of the program to run
+   * @param program the name of the program to run
+   * @param overrides the arguments to override in the program's configured user arguments before starting
+   * @param debug {@code true} if the program is to be started in debug mode, {@code false} otherwise
+   * @return {@link RunId}
+   * @throws ConflictException if the specified program is already running, and if concurrent runs are not allowed
+   * @throws NotFoundException if the specified program or the app it belongs to is not found in the specified namespace
+   * @throws IOException if there is an error starting the program
+   * @throws UnauthorizedException if the logged in user is not authorized to start the program. To start a program,
+   *                               a user requires {@link ApplicationPermission#EXECUTE} on the program
+   * @throws Exception if there were other exceptions checking if the current user is authorized to start the program
+   */
+  //TODO replace params with ProgramReference
+  @Deprecated
```

Comment:
I think it is the other run method we should deprecate? This one is running the latest version, we don't want people to use the other one to run old version programs

Reply:
right, the one that takes ProgramId is the run that should be deprecated